File: C/project-euler/Q001.c
/*
If we list all the natural numbers below 10 that are multiples of 3 or 5, we get
3, 5, 6 and 9. The sum of these multiples is 23.

Find the sum of all the multiples of 3 or 5 below 1000.
*/

#include <stdio.h>

int sum_multiples(int, int);

int main(void) {
  int a = sum_multiples(3, 1000);
  int b = sum_multiples(5, 1000);
  int c = sum_multiples(15, 1000);

  printf ("%d", a + b - c);

  return 0;
}

int sum_multiples(int n, int limit) {
  int i = n; // Backup
  int sum = 0;

  while (n < limit) {
    sum = sum + n;
    n += i;
  }

  return sum;
}

File: Haskell/project-euler/Q001.hs
{-
If we list all the natural numbers below 10 that are multiples of 3 or 5, we get 3, 5, 6 and 9. The sum of these multiples is 23.
Find the sum of all the multiples of 3 or 5 below 1000.
-}

multiples :: Int -> Int -> [Int]
multiples n limit = [n | n <- [n, n+n..limit-1]]

sum_multiples :: Int -> Int -> Int
sum_multiples n limit = sum (multiples n limit)

main = putStrLn $ show $ (sum_multiples 3 1000) + (sum_multiples 5 1000) - (sum_multiples 15 1000)
